### PR TITLE
Added BowerInstall command to install bower components of modules

### PR DESCRIPTION
**Pull request by mvdstam**

### CODE REVIEW

File: Commands/BowerInstallCommand.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        parent::__construct();
+
+        $this->modules = $modules;
+    }
+
+
+    public function fire()
+    {
+        // Check bower is installed
```
Comment (nWidart) [inline Commands/BowerInstallCommand.php]: Comment not needed. The `isBowerInstalled` is already clear enough.

However, maybe display an error message if bower is not installed, so the user knows what happened.

File: Commands/BowerInstallCommand.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            $module = $this->modules->findOrFail($module);
+
+            $this->handleModuleBowerInstall($module);
+        }
+    }
+
+
+    protected function handleModuleBowerInstall(Module $module)
+    {
+        // Check if bower.json exists
```
Comment (nWidart) [inline Commands/BowerInstallCommand.php]: Remove comment, and extract following 2 lines to descriptive method name. Something like `hasBowerFile()` & display error message.

File: Commands/BowerInstallCommand.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        $path = $module->getPath();
+        if (!file_exists($path . '/bower.json')) return;
+
+        // Run bower install
```
Comment (nWidart) [inline Commands/BowerInstallCommand.php]: Please remove comment. :)

File: Commands/BowerInstallCommand.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    /**
+     * @return boolean
+     */
+    protected function isBowerInstalled()
+    {
+        $result = null;
+        $output = [];
+        exec('bower', $output, $result);
+
+        if ($result === 127) {
```
Comment (nWidart) [inline Commands/BowerInstallCommand.php]: maybe extract to method so we know what `127` means in two months.

**Comment by mvdstam** [comment]
Thanks for the feedback, I'll take a look at it tomorrow.

File: Commands/BowerInstallCommand.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            $this->error('Bower is not installed, or could not be found.');
+            return false;
+        } elseif ($result !== 0) {
```
Comment (nWidart) [inline Commands/BowerInstallCommand.php]: Same here, something like `elseif (this->isUnkownError()`

File: Commands/BowerInstallCommand.php (hunk 1 above)
Comment (mvdstam) [inline Commands/BowerInstallCommand.php]: It already reports when Bower is not installed, or when something went wrong during the check itself, in `isBowerInstalled()`. No need for duplicate reports IMHO.

File: Commands/BowerInstallCommand.php (hunk 2 above)
Comment (mvdstam) [inline Commands/BowerInstallCommand.php]: Agreed, but not concerning the error message when a `bower.json` file wasn't found. Instead, let's do a simple `$this->info()` informing the user. Not all modules have bower-installed assets, and not using bower shouldn't result in an erroneous message.

Comment (nWidart) [inline Commands/BowerInstallCommand.php]: Ah yes, you're right about the message, info will suffice. :+1:

File: Commands/BowerInstallCommand.php (hunk 4 above)
Comment (mvdstam) [inline Commands/BowerInstallCommand.php]: Extracting two simple checks into two seperate methods is a bit over the top IMHO. I think descriptive constants containing the error codes are more appropriate here:

`if ($result === self::EXIT_COMMAND_NOT_FOUND)`

and

`if ($result !== self::EXIT_SUCCESS)`

That said, if you really want me to extract into methods; let me know.

Comment (nWidart) [inline Commands/BowerInstallCommand.php]: Constants can work too, though I like them slightly less.
Small methods of 1 line have a place too, they increase the code readability, without any performance loss.

**Comment by mvdstam** [comment]
Is this ready to merge, @nWidart? :smile:

**Pull request merged**